x/oracle/keeper/keeper.go

```diff
@@ -119,6 +122,7 @@ func (k Keeper) AppendValidatorToProphecy(ctx sdk.Context, networkDescriptor typ
 		return prophecy.Status, types.ErrProphecyFinalized
 
 	default:
+		// This is possible IFF GetProphecy returned with UNSPECIFIED
```

Comment:
I think we should ensure Prophecy Stored in Keeper never has UNSPECIFIED. UNSPECIFIED is 0, the default in golang if no value is passed when constructing Prophecy object

Reply:
I agree, the UNSPECIFIED as returned value is not good, we can add a new status enum. but it is not a big issue, the second value is an error, so it means UNSPECIFIED should not be used by caller.